### PR TITLE
Add priming logic to integration tests to avoid backend cold start issues.

### PR DESCRIPTION
**Pull request by mikelehen**

### CODE REVIEW

File: packages/firestore/test/integration/prime_backend.test.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as firestore from '@firebase/firestore-types';
+import { expect } from 'chai';
+import { EventsAccumulator } from './util/events_accumulator';
+import { withTestDoc } from './util/helpers';
+
+// Firestore databases can be subject to a ~30s "cold start" delay if they have not been used
```
[the remark this answers is not in the document]
Comment (mikelehen) [reply packages/firestore/test/integration/prime_backend.test.ts]: 93.  Last week I added tslint enforcement of 100 (https://github.com/firebase/firebase-js-sdk/blob/1221dba9ae05cec8ed002555871aab815e1f187b/packages/firestore/tslint.json#L28), so I'm leaving this as is unless you care strongly about changing it.

**Comment by mikelehen** [comment]
Yeah, A before() block outside the context of a describe() will run before all tests in all suites.

**Comment by wilhuff** [comment]
Doesn't this mean we start a listener and do a transaction for *every* test then? It seems like if all we care about is "priming" then we should be able to run this once.

Alternatively if test run times before and after are comparable then I guess it doesn't matter, but intuitively this seems like it could add up.

**Comment by mikelehen** [comment]
Sorry, I was trying to describe when it runs, not how often...  It'll only run once, regardless of how many tests are running.

**Pull request merged**